### ISSUE TITLE
Build fails on PHP 7.3 (beta)

### ISSUE THREAD

**Issue by Firehed** [tideways/php-xhprof-extension#61]
Hi,

I've been doing some prep work for the upcoming PHP 7.3 release, and found that the latest 4.x (4.1.6) release of the extension fails to build, emitting the following errors:

```
 cc -I. -I/tmp/php-xhprof-extension-4.1.6 -DPHP_ATOM_INC -I/tmp/php-xhprof-extension-4.1.6/include -I/tmp/php-xhprof-extension-4.1.6/main -I/tmp/php-xhprof-extension-4.1.6 -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/php-xhprof-extension-4.1.6/tideways.c  -fPIC -DPIC -o .libs/tideways.o
/tmp/php-xhprof-extension-4.1.6/tideways.c: In function 'tw_trace_callback_mongodb_connect':
/tmp/php-xhprof-extension-4.1.6/tideways.c:858:46: warning: passing argument 3 of 'tw_span_annotate_string' from incompatible pointer type [-Wincompatible-pointer-types]
         tw_span_annotate_string(idx, "host", url->host, 1 TSRMLS_CC);
                                              ^~~
In file included from /tmp/php-xhprof-extension-4.1.6/tideways.c:39:0:
/tmp/php-xhprof-extension-4.1.6/spans.h:6:6: note: expected 'char *' but argument is of type 'zend_string * {aka struct _zend_string *}'
 void tw_span_annotate_string(long spanId, char *key, char *value, int copy TSRMLS_DC);
      ^~~~~~~~~~~~~~~~~~~~~~~
/tmp/php-xhprof-extension-4.1.6/tideways.c: In function 'tw_pcre_match':
/tmp/php-xhprof-extension-4.1.6/tideways.c:1729:8: error: dereferencing pointer to incomplete type 'pcre_cache_entry {aka struct _pcre_cache_entry}'
     pce->refcount++;
        ^~
/tmp/php-xhprof-extension-4.1.6/tideways.c: In function 'hp_init_trace_callbacks':
/tmp/php-xhprof-extension-4.1.6/tideways.c:2684:20: warning: implicit declaration of function 'GC_G' [-Wimplicit-function-declaration]
     TWG(gc_runs) = GC_G(gc_runs);
                    ^~~~
/tmp/php-xhprof-extension-4.1.6/tideways.c:2684:25: error: 'gc_runs' undeclared (first use in this function)
     TWG(gc_runs) = GC_G(gc_runs);
                         ^~~~~~~
/tmp/php-xhprof-extension-4.1.6/tideways.c:2684:25: note: each undeclared identifier is reported only once for each function it appears in
/tmp/php-xhprof-extension-4.1.6/tideways.c:2685:30: error: 'collected' undeclared (first use in this function)
     TWG(gc_collected) = GC_G(collected);
                              ^~~~~~~~~
/tmp/php-xhprof-extension-4.1.6/tideways.c: In function 'hp_get_file_summary':
/tmp/php-xhprof-extension-4.1.6/tideways.c:2998:32: warning: format '%s' expects argument of type 'char *', but argument 5 has type 'zend_string * {aka struct _zend_string *}' [-Wformat=]
         snprintf(ret, len, "%s%s://", ret, url->scheme);
                                ^
/tmp/php-xhprof-extension-4.1.6/tideways.c:3005:32: warning: format '%s' expects argument of type 'char *', but argument 5 has type 'zend_string * {aka struct _zend_string *}' [-Wformat=]
         snprintf(ret, len, "%s%s", ret, url->host);
                                ^
/tmp/php-xhprof-extension-4.1.6/tideways.c:3013:32: warning: format '%s' expects argument of type 'char *', but argument 5 has type 'zend_string * {aka struct _zend_string *}' [-Wformat=]
         snprintf(ret, len, "%s%s", ret, url->path);
                                ^
/tmp/php-xhprof-extension-4.1.6/tideways.c: In function 'hp_stop':
/tmp/php-xhprof-extension-4.1.6/tideways.c:3814:19: error: 'gc_runs' undeclared (first use in this function)
         if ((GC_G(gc_runs) - TWG(gc_runs)) > 0) {
                   ^~~~~~~
/tmp/php-xhprof-extension-4.1.6/tideways.c:3816:50: error: 'collected' undeclared (first use in this function)
             tw_span_annotate_long(0, "gcc", GC_G(collected) - TWG(gc_collected) TSRMLS_CC);
                                                  ^~~~~~~~~
/tmp/php-xhprof-extension-4.1.6/tideways.c: In function 'zif_tideways_span_callback':
/tmp/php-xhprof-extension-4.1.6/tideways.c:4128:36: error: lvalue required as increment operand
             GC_REFCOUNT(fci.object)++;
                                    ^~
make: *** [Makefile:197: tideways.lo] Error 1
```

Obviously some stuff may change between now and the actual 7.3 release, but there probably won't be significant changes. It builds without any errors or warnings through 7.2.9.

Build environment: Official PHP 7.3-rc-fpm-alpine Docker image:

```Dockerfile
FROM php:7.3-rc-fpm-alpine as env
RUN apk add --update \
        $PHPIZE_DEPS \
        icu-dev \
        postgresql-dev \
        zlib-dev \
    && pecl install apcu \
    && docker-php-ext-enable apcu \
    && docker-php-ext-install intl \
    && rm -rf /var/cache/apk/*
ENV XHPROF_VERSION=4.1.6
RUN wget https://github.com/tideways/php-xhprof-extension/archive/v$XHPROF_VERSION.tar.gz -O /tmp/xhprof.tar.gz \
    && cd /tmp \
    && tar -xzvf xhprof.tar.gz \
    && cd /tmp/php-xhprof-extension-$XHPROF_VERSION \
    && phpize \
    && ./configure \
    && make \
    && make install \
    && cd - \
    && rm /tmp/xhprof.tar.gz \
    && rm -rf /tmp/php-xhprof-extension-$XHPROF_VERSION \
    && echo 'extension=tideways.so' > /usr/local/etc/php/conf.d/tideways.ini
# rest removed, not relevant
```

**Comment by beberlei** [comment]
@Firehed thanks for opening the issue, i have this on my schedule. It is a simple fix, the GC stats API changed between 7.2 and 7.3 and this just needs to be handled.

**Comment by beberlei** [comment]
@Firehed i pushed a commit for 7.3 support.
